external/upstream/pybind11/CMakeLists.txt

```diff
@@ -16,7 +16,6 @@ else()
         UPDATE_COMMAND ""
         CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${STAGED_INSTALL_PREFIX}
                    -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
-                   -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}  # unused, but needs working compiler
```

Comment:
This one came about when there's a C compiler available (say, `/usr/bin/gcc` 4.6) but not up to compiling anything serious. Errors arise when cmake tries to compile a simple test prog, even though a better C compiler has been passed in through `CMAKE_C_COMPILER`.

Reply:
I removed it because of the annoying "Passed variable not used by project" warning. The culprit for the need of a C compiler seems to be the check for `PThreads` in pybind11 own `CMakeLists.txt`: https://github.com/pybind/pybind11/blob/master/CMakeLists.txt#L15-L20
According to the comment there, problem is solved with CMake higher than 3.4.0 That would explain why I never saw the problem myself. Mind if I bump the CMake requirement to 3.8?

Reply:
I think we're safe, policy-wise https://cmake.org/cmake/help/v3.8/manual/cmake-policies.7.html#policies-introduced-by-cmake-3-4